src/sentry/ingest/userreport.py

```diff
@@ -93,10 +107,119 @@ def find_event_user(report_data, event):
             return None
 
     tag = event.get_tag("sentry:user")
+
     if not tag:
         return None
 
     try:
         return EventUser.for_tags(project_id=report_data["project_id"], values=[tag])[tag]
     except KeyError:
         pass
+
+
+def find_event_user_with_snuba(event: Event, eventuser: EventUser):
+    """
+    Query Snuba to get the EventUser information for an Event.
+    Then compare with the EventUser record and confirm that the
+    query result is equivalent.
+    """
+    start_date, end_date = _start_and_end_dates(event.datetime)
```

Comment:
Nice idea here, this will make querying much cheaper

Reply:
I actually just realised, can we just the `event` directly here since we already have it?  It should also have the user information, and be the same as what we have in Snuba